### PR TITLE
[PAPI-60] Market History

### PR DESCRIPTION
**Pull request by mrtpain**
- Filter history by start and end date for daily data
- Improve snapshots for transactions
    - Performance improvements using dictionary preventing multiple lookups
    - Snapshot reserves logs improving snapshot accuracy

### CODE REVIEW

File: src/Opdex.Platform.Application/EntryHandlers/LiquidityPools/Snapshots/ProcessLiquidityPoolSnapshotsByTransactionCommandHandler.cs [hunk 1]
```diff
@@ -31,39 +38,88 @@ public ProcessLiquidityPoolSnapshotsByTransactionCommandHandler(IMediator mediat
...
-            var crsSnapshotQuery = new RetrieveTokenSnapshotWithFilterQuery(crsToken.Id, 0, blockTime, SnapshotType.Minute);
-            var crsSnapshot = await _mediator.Send(crsSnapshotQuery);
+                    var srcToken = await _mediator.Send(new RetrieveTokenByAddressQuery(createPoolLog.Token));
+                    tokens[srcToken.Id] = srcToken;
 
-            if (crsSnapshot.Id == 0 || crsSnapshot.EndDate < blockTime)
-            {
-                await _mediator.Send(new CreateCrsTokenSnapshotsCommand(blockTime, block.Height));
-                crsSnapshot = await _mediator.Send(crsSnapshotQuery);
-            }
```
Comment (mrtpain) [inline src/Opdex.Platform.Application/EntryHandlers/LiquidityPools/Snapshots/ProcessLiquidityPoolSnapshotsByTransactionCommandHandler.cs:45]: Might bring this back as a "just incase"

File: src/Opdex.Platform.Application.Abstractions/EntryQueries/Markets/GetMarketSnapshotsWithFilterQuery.cs [hunk 2]
```diff
@@ -1,27 +1,32 @@
...
+        /// </summary>
+        /// <param name="market">The address of the market.</param>
+        /// <param name="cursor">The snapshot cursor filter.</param>
+        public GetMarketSnapshotsWithFilterQuery(Address market, SnapshotCursor cursor)
         {
-            if (marketAddress == Address.Empty)
+            if (market == Address.Empty)
             {
-                throw new ArgumentNullException(nameof(marketAddress), $"{nameof(marketAddress)} must be provided.");
+                throw new ArgumentNullException(nameof(market), "Liquidity pool address must not be empty.");
```
Comment (drmathias) [inline src/Opdex.Platform.Application.Abstractions/EntryQueries/Markets/GetMarketSnapshotsWithFilterQuery.cs:22]: nitpick: Message says liquidity pool but the param is market

File: src/Opdex.Platform.Application/EntryHandlers/Markets/Snapshots/GetMarketSnapshotsWithFilterQueryHandler.cs [hunk 3]
```diff
@@ -26,19 +25,19 @@ public GetMarketSnapshotsWithFilterQueryHandler(IMediator mediator, IMapper mapp
             _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
         }
 
-        public async Task<IEnumerable<MarketSnapshotDto>> Handle(GetMarketSnapshotsWithFilterQuery request, CancellationToken cancellationToken)
+        public override async Task<MarketSnapshotsDto> Handle(GetMarketSnapshotsWithFilterQuery request, CancellationToken cancellationToken)
         {
-            var market = await _mediator.Send(new RetrieveMarketByAddressQuery(request.MarketAddress));
+            var pool = await _mediator.Send(new RetrieveMarketByAddressQuery(request.Market), cancellationToken);
```
Comment (drmathias) [inline src/Opdex.Platform.Application/EntryHandlers/Markets/Snapshots/GetMarketSnapshotsWithFilterQueryHandler.cs:30]: Pool or market?

**Review by drmathias — COMMENTED** [review]
Nice 👍

File: src/Opdex.Platform.Application/EntryHandlers/Markets/Snapshots/GetMarketSnapshotsWithFilterQueryHandler.cs (hunk 3 above)
Comment (mrtpain) [reply src/Opdex.Platform.Application/EntryHandlers/Markets/Snapshots/GetMarketSnapshotsWithFilterQueryHandler.cs:30]: copy and paste 😆

**Pull request merged**